faiss/IndexHNSW.cpp

```diff
@@ -948,7 +948,7 @@ void IndexHNSWCagra::search(
 
             std::random_device rd;
             std::mt19937 gen(rd());
-            std::uniform_int_distribution<idx_t> distrib(0, this->ntotal);
+            std::uniform_int_distribution<idx_t> distrib(0, this->ntotal - 1);
```

Comment:
can this->ntotal be 0?

Reply:
Only if vectors have not been added to the index, then calling `search` is invalid in any case.